src/Data/Binary/Get/Internal.hs

```diff
@@ -415,5 +416,5 @@ unsafeReadN !n f = C $ \inp ks -> do
 
 readNWith :: Int -> (Ptr a -> IO a) -> Get a
 readNWith n f = do
-    readN n $ \s -> B.inlinePerformIO $ B.unsafeUseAsCString s (f . castPtr)
+    readN n $ \s -> accursedUnutterablePerformIO $ B.unsafeUseAsCString s (f . castPtr)
```

Comment:
So I think this is safe as long as the action is deterministic and has no external effects. It depends on the value of the bytestring so the value dependencies look ok.